test/integration/authentication_test.go

```diff
@@ -58,6 +41,8 @@ func TestKubernetesAuthentication(t *testing.T) {
 		rootCmd := cmd.GetRootCommand()
 		rootCmd.SetArgs([]string{
 			"start",
+			"--port",
+			strconv.Itoa(gatewayPort),
```

Comment:
Ideally we should be able to set `port=0` like `httptest.Server` to pick a random port to avoid port collision. Currently we can't do that because we don't have a way to retrieve that random port value. Hence, I hardcore the port value (`8443` and `8444`) for these tests.